runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -46,6 +46,21 @@ IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
   }
 }
 
+IntervalSet::~IntervalSet()
+{
+}
+
+IntervalSet& IntervalSet::operator=(const IntervalSet& other)
+{
+  if (_readonly) {
+    throw IllegalStateException("can't alter read only IntervalSet");
+  }
```

Comment:
Hmm, I'm not sure if that's the whole story. Way too many side effects possible...

We only have the runtime tests (there are a few performance tests in them, but rather for measuring of the overall process, hard to use them as comparison). At least the runtime could indicate if something goes wrong in general. You can easily run the C++ runtime tests only on your local box to check this.